src/BasicAccessFactory.php

```diff
@@ -17,10 +17,11 @@ class BasicAccessFactory
 {
     public function __invoke(ContainerInterface $container) : BasicAccess
     {
+        /** @var UserRepositoryInterface|\Mezzio\Authentication\UserRepositoryInterface|null $userRegister */
         $userRegister = $container->has(UserRepositoryInterface::class)
```

Comment:
You might want to require `webmozart/assert` and use `Assert::nullOrIsInstanceOfAny()` instead of inline annotation 👍

Reply:
It will be BC change, isn't it?

webmozart/assert will throw \InvalidArgumentException
In current code, we throw own package exception Mezzio\Authentication\Exception\InvalidConfigException

Reply:
At this stage, it isn't a BC break: we are silently assuming the type, but that's unsafe already.